module/pollSubmissions/helper.js

```diff
@@ -0,0 +1,324 @@
+            }
+        });
+    }
+
+
+    /**
+    * List all the polls.
+    * @method
+    * @name list
+    * @param {String} userId - userId
+    * @returns {JSON} - Polls list.
+    */
+
+   static list(userId= "") {
+    return new Promise(async (resolve, reject) => {
+        try {
+
+            if(userId == ""){
+                throw new Error(messageConstants.apiResponses.USER_ID_REQUIRED_CHECK);
+            }
+
+            let getPollList = [
+                pollsHelper.list
+                ( 
+                    userId
+                ),
+                database.models.pollSubmissions.aggregate
+                    (
+                        [
+                            { "$match": { "userId": userId, isDeleted: false } },
+                            {
+                                "$project": {
+                                    '_id': "$pollId",
+                                    'name': '$pollName'
+                                }
+                            },
+                            { "$sort": { createdAt: 1 } }
+                        ])
+            ];
+
+
+                let result = [];
+
+                await Promise.all(getPollList)
+                    .then(function (response) {
+                          
+                        if (Array.isArray(response[0].data) && response[0].data.length > 0) {
+                          result = result.concat(response[0].data);
+                        }
+
+                        if (response[1].length > 0) {
+                          result = result.concat(response[1]);
+                        }
+                    });
+
+
+                if (!result.length) {
+                    throw new Error(messageConstants.apiResponses.POLL_NOT_FOUND)
+                }
+
+                return resolve({
+                    success: true,
+                    message: messageConstants.apiResponses.POLLS_LIST_FETCHED,
+                    data: _.uniqWith(result, _.isEqual)
+                });
+
+            } catch (error) {
+                return resolve({
+                    success: false,
+                    message: error.message,
+                    data: false
+                });
+            }
+        });
+    }
+
+
+
+    /**
+    * Validate poll submission.
+    * @method
+    * @name validatePollSubmission
+    * @param {String} pollId - pollId
+    * @param {String} userId - userId
+    * @returns {String} - poll status and poll submission document.
+    */
+
+   static validatePollSubmission(pollId= "", userId= "") {
+    return new Promise(async (resolve, reject) => {
+        try {
+            
+            if (pollId == "") {
+                throw new Error(messageConstants.apiResponses.POLL_ID_REQUIRED_CHECK)
+            }
+
+            if (userId == "") {
+                throw new Error(messageConstants.apiResponses.USER_ID_REQUIRED_CHECK)
+            }
+
+            let pollDocument = await pollsHelper.pollDocuments
```

Comment:
@deepakr123 - Suggest creating a helper function in polls which tells given a pollId whether or not it is still open for submission or not?

Reply:
resolved